Documentation

nicolejkeeney opened the pull request:
This PR sets up a framework for improving and fleshing out documentation of climakitae. It's still a bit bare-bones now, and I have yet to add the entire section on scientific toolkits (i.e how to use the various explore panels, timeseries stuff, etc). Let me know what you all think of the documentation at this stage

In order to see how the readthedocs page will look (current version [here](https://climakitae.readthedocs.io/en/latest/)), you will need to **build the documentation locally**. You are welcome to do this, and there's instructions for how to do so [here](https://github.com/cal-adapt/climakitae/blob/documentation/docs/README.md). This is the only way you'll be able to preview how the readthedocs site will look before we merge this branch into main. 

If you'd like to get a sense for how this page will look but are uninterested in building the docs locally, I've provided some screenshots at the bottom. 

**How to review**: 
1) Observe (within GitHub is fine) the markdown in the "Working with our Data" rst file: https://github.com/cal-adapt/climakitae/blob/documentation/docs/data.rst
2) Provide some feedback on the design, either by [building the docs yourself locally](https://github.com/cal-adapt/climakitae/blob/documentation/docs/README.md) or just observing the screenshots included in below. 

**Homepage:**
<img width="1000" alt="Homepage" src="https://user-images.githubusercontent.com/66140951/214399562-54888159-4a19-422b-ad02-147f2b2a4e20.png">

**About our data:**
<img width="1000" alt="data" src="https://user-images.githubusercontent.com/66140951/214399714-10a20456-10bc-4de1-857c-b23dbfdb7df4.png">

**climakitae API:**
<img width="1000" alt="API page" src="https://user-images.githubusercontent.com/66140951/214399779-3074a2b4-6f6b-4ac8-b882-01e317e0f8e2.png">

**example of function documentation:**
<img width="1000" alt="documentation example" src="https://user-images.githubusercontent.com/66140951/214399868-fe18a1ec-d606-4362-890c-7c3b7b91f606.png">

elehmer reviewed (approved):
This all looks good.

vicford reviewed (changes requested):
This looks **amazing**, @nicolejkeeney, really excited to see how this builds upon in the future!

A couple of things, mainly questions: 
1. building locally didn't work for me, so just reviewing based on the docs provided in the PR
2. In the first instance where app = ck.Application there are no parentheses on the Application object (in the app.select description), but clearly shows in the core functions page, and what we do on the Hub too. If it doesn't make a difference (which upon testing on the hub, it does), I do think we should still be consistent about how we display that call in our docs. 
3. In "Directly modifying the location and selections attributes" section where things are highlighted, is it "selectors" or "selections"?
4. In "Using a csv config file", we appropriately state that the headers must be exactly as they are in the following example, meaning that they must be labeled precisely as such, and/or the order must follow that format? Can be more explicit here on this

nicolejkeeney commented:
> 2. In the first instance where app = ck.Application there are no parentheses on the Application object (in the app.select description), but clearly shows in the core functions page, and what we do on the Hub too. If it doesn't make a difference (which upon testing on the hub, it does), I do think we should still be consistent about how we display that call in our docs.

Good point, I added back in some parentheses and I think I fixed the issue! Thanks for spotting that.

nicolejkeeney commented:
> 3. In "Directly modifying the location and selections attributes" section where things are highlighted, is it "selectors" or "selections"?

It is "selections", which is a DataSelector object-- a bit confusing I guess! But selections is correct.

nicolejkeeney commented:
> 4. In "Using a csv config file", we appropriately state that the headers must be exactly as they are in the following example, meaning that they must be labeled precisely as such, and/or the order must follow that format? Can be more explicit here on this

I agree that that section is confusing. I modified the wording a bit to see if that cleared it up?

vicford reviewed (approved):
Looks great!

gdicecco commented on docs/data.rst:
```diff
@@ -0,0 +1,146 @@
+
+Retrieve and subset the data
+#############################
+In this section we will detail the various methods to retrieve and subset the catalog data. 
```
Here I suggest we either link to the data catalog page on the website (https://analytics.cal-adapt.org/data/) or link & condense some of that info here to make this able to stand on its own (i.e. mention somewhere what kind of data is in the catalog)

nicolejkeeney replied:
great idea!

gdicecco commented on docs/data.rst:
```diff
@@ -0,0 +1,146 @@
+If you are working in a Jupyter notebook environment, you can view and set your data and location 
+options in the :py:func:`climakitae.Application.select()` GUI (graphical user interface). This GUI also provides a visual overview of the various 
+datasets available in the AE data catalog. Using this GUI, you can chose what dataset you'd like to 
+retrieve-- chosing a variable, timeslice, resolution, etc.-- and the location for which you'd like to 
```
Typo - 'choosing a variable...'

gdicecco commented on docs/data.rst:
```diff
@@ -0,0 +1,146 @@
+
+Read the data into memory 
+###########################
+The data is retrieved as lazily loaded Dask arrays until you chose to read the data into 
```
typo *choose to read....

gdicecco commented on docs/data.rst:
```diff
@@ -0,0 +1,146 @@
+################
+To export the data, first chose the filetype you want to export the data to using the 
+:py:func:`climakitae.Application.export_as()` dropdown menu. This will allow you to choose 
+between three options: netcdf, csv, and geotiff.::
```
formatting thing - I don't think we need the . after geotiff here?

gdicecco commented on docs/data.rst:
```diff
@@ -0,0 +1,146 @@
+
+   app.export_as() 
+
+After selecting your desired output filetype, input the data you want to export and the 
```
To be thorough here, it would be nice to repeat some of the markdown Beth wrote describing what data structures are suited to each export type (at a minimum that netcdf is most flexible, but csv is very restricted).

nicolejkeeney replied:
good idea!

gdicecco reviewed (approved):
This is an awesome start to the documentation! A few specific comments are below.

More broadly, I'm usure how I feel about the link to the Getting Started nb render on the website being featured prominently here - I think it might be hard to parse that this is from a notebook that should be run elsewhere. But, maybe this is a good thing to have for now and later instead we would want to link to e.g. a binder instance of the getting started notebook or something like that.
